Model/Sync/Catalog/Processor.php

```diff
@@ -844,4 +886,72 @@ private function updateSyncTable($syncDataRecord)
             [$syncDataRecord]
         );
     }
+
+    /**
+     * @param int $parentId
+     * @param array <mixed> $parentItemsData
+     * @param array <mixed> $yotpoSyncTableItemsData
+     * @param array <mixed> $yotpoFormatItemData
+     * @return string|null
+     */
+    private function ensureEntityExistenceAsProductInYotpo(
+        $parentId,
+        $parentItemsData,
+        $yotpoSyncTableItemsData,
+        $yotpoFormatItemData
+    ) {
+        $parentProductsData = $this->getCollectionForSync([$parentId])->getItems();
+        if (!count($parentProductsData)) {
+            return null;
+        }
+        $apiRequestParams = $this->getApiParams(
+            $parentId,
+            $yotpoSyncTableItemsData,
+            [],
+            $parentItemsData,
+            false
+        );
+        $method = $apiRequestParams['method'];
+        if (!$method) {
+            return null;
+        }
```

Comment:
Hi @manaftryzens, why do we need this check?
`$method` should never be `null` because all the edge cases [were removed](https://github.com/YotpoLtd/magento2-module-core/pull/138/commits/e11fcf5247c704ff8c01a31fea90c98e17376493#diff-efe4378f95b0c72fc3ade0f21aec60cc7745852a8e494652f130deea5dd18ebdL409).